src/mumble/ALSAAudio.cpp

```diff
@@ -373,27 +375,15 @@ void ALSAAudioInput::run() {
 
 	qml.unlock();
 
-	while (bRunning) {
+	while (bRunning && bOk) {
 #ifdef ALSA_VERBOSE
 		snd_pcm_status_malloc(&status);
 		snd_pcm_status(capture_handle, status);
 		snd_pcm_status_dump(status, log);
 		snd_pcm_status_free(status);
 #endif
-		readblapp = snd_pcm_readi(capture_handle, inbuff, static_cast<int>(wantPeriod));
-		if (readblapp == -ESTRPIPE) {
-			qWarning("ALSAAudioInput: PCM suspended, trying to resume");
-			while (bRunning && snd_pcm_resume(capture_handle) == -EAGAIN)
-				msleep(1000);
-			if ((err = snd_pcm_prepare(capture_handle)) < 0)
-				qWarning("ALSAAudioInput: %s: %s", snd_strerror(static_cast<int>(readblapp)), snd_strerror(err));
-		} else if (readblapp == -EPIPE) {
-			err = snd_pcm_prepare(capture_handle);
-			qWarning("ALSAAudioInput: %s: %s", snd_strerror(static_cast<int>(readblapp)), snd_strerror(err));
-		} else if (readblapp < 0) {
```

Comment:
snd_pcm_recover doesn't handle this case.

Per the docs, it could happen due to -EBADFD. For that, it seems like snd_pcm_prepare is the right thing to do?